records_mover/db/postgres/copy_options/date_input_style.py

```diff
@@ -0,0 +1,251 @@
+from records_mover.utils import quiet_remove
+from records_mover.records.hints import cant_handle_hint
+from records_mover.records.types import RecordsHints
+from typing import Optional, Set, Union, TYPE_CHECKING
+if TYPE_CHECKING:
+    from typing import Literal
+    # https://www.postgresql.org/docs/9.5/runtime-config-client.html#GUC-DATESTYLE
+    #
+    # DateStyle (string)
+    #
+    #  Sets the display format for date and time values, as well as
+    #  the rules for interpreting ambiguous date input values. For
+    #  historical reasons, this variable contains two independent
+    #  components: the output format specification (ISO, Postgres,
+    #  SQL, or German) and the input/output specification for
+    #  year/month/day ordering (DMY, MDY, or YMD). These can be set
+    #  separately or together. The keywords Euro and European are
+    #  synonyms for DMY; the keywords US, NonEuro, and NonEuropean are
+    #  synonyms for MDY. See Section 8.5 for more information. The
+    #  built-in default is ISO, MDY, but initdb will initialize the
+    #  configuration file with a setting that corresponds to the
+    #  behavior of the chosen lc_time locale.
+    DateInputStyle = Union[Literal["DMY"], Literal["MDY"]]
+else:
+    DateInputStyle = str
+
+
+def determine_date_input_style(unhandled_hints: Set[str],
```

Comment:
C901 'determine_date_input_style' is too complex (16)

Reply:
It's true.  I suppose I could break this up into individual functions for the different hints; instead I made some cleanups elsewhere to get under (and ratchet) the quality threshold.